core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessageDuplicator.java

```diff
@@ -74,19 +77,38 @@
      * @param publisher the publisher who will publish data to subscribers
      */
     protected AbstractStreamMessageDuplicator(U publisher) {
+        this(publisher, obj -> 0);
+    }
+
+    /**
+     * Creates a new instance wrapping a {@code publisher} and publishing to multiple subscribers.
+     * @param publisher the publisher who will publish data to subscribers
+     * @param signalLengthGetter the signal length getter that is used to limit the total length of signals
+     */
+    protected AbstractStreamMessageDuplicator(U publisher, Function<Object, Integer> signalLengthGetter) {
```

Comment:
Could we define a dedicated interface for this? Otherwise it will do auto-boxing pretty often.